cpp_linter/run.py

```diff
@@ -266,7 +266,7 @@ def set_exit_code(override: int = None) -> int:
         then this value will describe (like a bool value) if any checks failed.
     """
     exit_code = override if override is not None else bool(Globals.OUTPUT)
-    print(f"::set-output name=checks-failed::{exit_code}")
+    os.environ["GITHUB_OUTPUT"] = f"checks-failed={exit_code}"
```

Comment:
I'm not sure if this will work with GitHub's runner context. [The GitHub docs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) say we should write/append the var (as a `key=value`) to a file pointed to by `GITHUB_ENV`.

Reply:
I'm trying to say that the step's env is flushed when the step finishes. That's why github recommends writing to a "env file" so the env change can persist across steps.
```python
    try:
        with open(os.environ["GITHUB_OUTPUT"], "a", encooding="utf-8")as env_file:
            env_file.write(f"checks-failed={exit_code}")  # maybe add `\n` if using multiple vars?
    except FileNotFoundError:
        # not executed on a github CI runner; ignore this error when executed locally
        pass
```

Reply:
The examples from that page are still `echo`ing to a file (`>>`) - not stdout.